examples/systemcalls/systemcalls.c

```diff
@@ -82,8 +117,40 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     command[count] = NULL;
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
-    command[count] = command[count];
-
+    //command[count] = command[count];    
+    
+    int pid, fd, status;
+    
+    //Check if the outputfile path provided exists and opens
+    fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644); //opens the file specified by argument 'outputfile' with rwx permissions for user, and rx permissions for group and others
+    if(fd<0){
+    	perror("Unable to open file. Exiting...");
+    	abort();
+    }
+    //replace stdout with file provided. stdout has fd = 1
+    dup2(fd, 1);
+    
+    pid = fork();
+    if(pid == -1)
+    	return -1; //failed to fork
+    else if(pid == 0){
+    	execv(command[0], command);
+    	exit(-1);
+    }
+    if(waitpid(pid, &status, 0) == -1)
+    	return false;
+    else if (WIFEXITED(status)){
+    	int exit_status = WEXITSTATUS(status); 
+    	if(exit_status == status)
+    		return true;
+    	else
+    		return false;    	
+    }  	
+    
+    va_end(args);    
```

Comment:
@prvbharadwaj1731 think about how you can avoid dupicating logic between these two do_exec functions.